.github/action_scripts/check_clusters.js

```diff
@@ -0,0 +1,218 @@
+const fetch = require( 'node-fetch' );
+
+const fetchData = async ( url ) => {
+    try {
+        const response = await fetch( url, {
+            method: 'GET',
+            headers: {
+                Accept: 'application/json'
+            }
+        } );
+
+        const responseParsed = await response.json();
+
+        if( response.status !== 200 ) {
+            throw Error( `Response status from URL: ${url} is not 200` );
+        }
+
+        return responseParsed;
+    } catch( e ) {
+        throw Error( `Error when fetching data: ${e.message}` );
+    }
+};
+
+const sleep = ( ms ) => {
+    return new Promise( ( resolve ) => setTimeout( resolve, ms ) );
+};
+
+const checkIfNodeIsReady = async ( url, name ) => {
+    console.log( `Checking if ${name} is ready` );
+    const checkInterval = 10 * 1000;
+    for( let idx = 0; idx < 12; idx++ ) {
+        const { state } = await fetchData( url );
+        if( state === 'Ready' ) {
+            console.log( `Node ${name} is ready` );
+            return;
+        }
+        console.log(
+            `Node ${name} is not ready yet, waiting ${checkInterval / 1000}s`
+        );
+        await sleep( checkInterval );
+    }
+
+    throw Error(
+        `Node ${name} is not ready after ${
+            ( checkInterval * 12 ) / 1000
+        }s, check the logs.`
+    );
+};
+
+const validateOrdinalsAndSnapshots = async ( urls ) => {
+    const ordinals = [];
+    for( const url of urls ) {
+        const snapshot = await fetchData( `${url}/latest` );
+        ordinals.push( snapshot.value.ordinal );
+    }
```

Comment:
I don't know how much you're into functional programming but you can use that array with URLs and using `Promise.all` get responses that you can map then to snapshot ordinals like:

```
.then(_ => _.value.ordinal)
```

Reply:
Hmm, good catch. I know this functionality and maybe I've missed this part, I'll update =].

Reply:
in general, FP approach is that you avoid mutations and side effects.
```
const ordinals = []
...
ordinals.push(...)
```
is a mutation. in FP, you read the code as small programs and knowing that all structures are immutable, it's just easier for local reasoning and understanding the flow of the code. I'm not sure if you want to follow FP in the code but this is something we rely on in the protocol, so just instilling the idea :)